spec.html

```diff
@@ -4278,73 +4278,91 @@ <h1>UpdateEmpty ( _completionRecord_, _value_ )</h1>
-        </emu-alg>
-      </emu-clause>
+      <emu-table id="table-reference-fields" caption="Reference Fields">
+        <table>
+          <tbody>
+            <tr>
+              <th>Field Name</th>
+              <th>Value</th>
+              <th>Meaning</th>
+            </tr>
+            <tr>
+              <td oldids="sec-getbase,ao-getbase">[[Base]]</td>
+              <td>
+                One of:
+                <ul>
+                  <li>
+                    ECMAScript language value except *undefined* and *null*;
+                  </li>
+                  <li>
+                    Environment Record;
+                  </li>
+                  <li>
+                    ~empty~.
+                  </li>
+                </ul>
+              </td>
+              <td>~empty~ value indicates that the Reference could not be resolved to a binding.</td>
+            </tr>
+            <tr>
+              <td oldids="sec-getreferencedname,ao-getreferencedname">[[ReferencedName]]</td>
+              <td>String or Symbol</td>
+              <td>Binding name. Always a String if [[Base]] value is an Environment Record.</td>
+            </tr>
+            <tr>
+              <td oldids="sec-isstrictreference,ao-isstrictreference">[[Strict]]</td>
+              <td>Boolean</td>
+              <td>*true* if the Reference originated in strict mode code, *false* otherwise.</td>
+            </tr>
+            <tr>
+              <td>[[ThisValue]]</td>
+              <td>any ECMAScript language value or ~empty~</td>
+              <td>If not ~empty~, the Reference represents a name binding that was expressed using the `super` keyword; it is called a <dfn id="super-reference">Super Reference</dfn> and its [[Base]] value will never be an Environment Record.</td>
+            </tr>
+          </tbody>
+        </table>
+      </emu-table>
 
-      <emu-clause id="sec-isstrictreference" aoid="IsStrictReference" oldids="ao-isstrictreference">
-        <h1>IsStrictReference ( _V_ )</h1>
-        <p>The abstract operation IsStrictReference takes argument _V_. It performs the following steps when called:</p>
-        <emu-alg>
-          1. Assert: Type(_V_) is Reference.
-          1. Return the strict reference flag of _V_.
-        </emu-alg>
-      </emu-clause>
+      <p>The following abstract operations are used in this specification to operate upon References:</p>
 
       <emu-clause id="sec-hasprimitivebase" aoid="HasPrimitiveBase" oldids="ao-hasprimitivebase">
         <h1>HasPrimitiveBase ( _V_ )</h1>
         <p>The abstract operation HasPrimitiveBase takes argument _V_. It performs the following steps when called:</p>
         <emu-alg>
-          1. Assert: Type(_V_) is Reference.
-          1. If Type(_V_'s base value component) is Boolean, String, Symbol, BigInt, or Number, return *true*; otherwise return *false*.
+          1. Assert: _V_ is a Reference Record.
+          1. If _V_.[[Base]] is ~empty~, then return *false*.
```

Comment:
Nit: we generally say `If foo, return *false*.` rather than `If foo, then return *false*.` I'll add a lint rule for this at some point.

Reply:
Fixed in ae8fb93. A lint rule would be great: there are a few `If foo, then return *false*.` occurrences remaining.

Reply:
PR to add lint rule: https://github.com/tc39/ecmarkup/pull/264